tests/core_tests/service_nodes.cpp

```diff
@@ -123,30 +123,31 @@ bool gen_service_nodes::generate(std::vector<test_event_entry> &events) const
 {
 
   linear_chain_generator gen(events);
-  gen.create_genesis_block();                           //  1
+  gen.create_genesis_block();
 
   const auto miner = gen.first_miner();
   const auto alice = gen.create_account();
 
-  gen.rewind_until_v9();                                //  3
-  gen.rewind_blocks_n(10);                              // 13
+  const get_test_options<gen_service_nodes> test_options = {};
+  gen.rewind_until_version(test_options.hard_forks, network_version_9_service_nodes);
+  gen.rewind_blocks_n(10);
 
-  gen.rewind_blocks();                                  // 13 + N
+  gen.rewind_blocks();
 
   const auto tx0 = gen.create_tx(miner, alice, MK_COINS(101));
-  gen.create_block({tx0});                              // 14 + N
+  gen.create_block({tx0});
 
-  gen.rewind_blocks();                                  // 14 + 2N
+  gen.rewind_blocks();
 
   const auto reg_tx = gen.create_registration_tx(alice, m_alice_service_node_keys);
 
-  gen.create_block({reg_tx});                           // 15 + 2N
+  gen.create_block({reg_tx});
 
   DO_CALLBACK(events, "check_registered");
 
   for (auto i = 0u; i < service_nodes::get_staking_requirement_lock_blocks(cryptonote::FAKECHAIN); ++i) {
     gen.create_block();
-  } // 15 + 2N + M
```

Comment:
Should probably just set `count` in `gen_service_nodes::check_expired` to a large number if we don't want to track the number of blocks manually.